### ISSUE TITLE
V1.8.13 fluentbit stops sending logs ~10 mins when ingesting 100,000 logs per second

### ISSUE THREAD

**Issue by sophieyfang** [fluent/fluent-bit#4991]
## Bug Report

**Describe the bug**

V1.8.13 fluentbit stops sending logs ~10 mins when ingesting 100,000 logs per second

**To Reproduce**

```
Step 1: run log_generator
root      8867 32.9  0.0  [47688 10748](tel:47688 10748) ?        R    Mar03 411:00 python3 /log_generator.py --log-size-in-bytes 1000 --log-rate 100000 --log-write-type file --file-path /tail/tail_log
Step 2: run fluentbit
sophieyfang_google_com@perf-sophieyfang-5-1:~$ ps aux | grep fluent
root      5294  0.0  0.0 173488  6484 ?        S    Mar03   0:00 sudo nohup /fluent-bit/build/bin/fluent-bit -c /fluent-bit-config.conf
root      5296  3.4  0.4 [1006688 289476](tel:1006688 289476) ?      Sl   Mar03  42:42 /fluent-bit/build/bin/fluent-bit -c /fluent-bit-config.conf
root      8866  0.0  0.0 173488  6528 ?        S    Mar03   0:00 sudo nohup python3.7 /collect_fluent_bit_metrics.py
root      8868  0.0  0.0 [567800 47780](tel:567800 47780) ?        Sl   Mar03   1:07 python3.7 /collect_fluent_bit_metrics.py
sophiey+ 20179  0.0  0.0  14860  1112 pts/0    S+   16:59   0:00 grep --color=auto fluent
sophieyfang_google_com@perf-sophieyfang-5-1:~$ cat /fluent-bit-config.conf 
[SERVICE]
    flush        1
    daemon       Off
    log_level    debug
    parsers_file parsers.conf
    plugins_file plugins.conf
    HTTP_Server  On
    HTTP_Listen  0.0.0.0
    HTTP_PORT    2020
    storage.metrics           on
    storage.path              /var/log/flb-storage/
    storage.sync              normal
    storage.checksum          off
    storage.backlog.mem_limit 1G
    storage.max_chunks_up     512
[INPUT]
    Name          tail
    Alias         input_main
    Tag           perf-benchmark-test
    Path          /tail/tail_log
    storage.type  filesystem
    Buffer_Chunk_Size   32000
    Buffer_Max_Size     32000
[OUTPUT]
    Name        stackdriver
    storage.total_limit_size    [10737418240](tel:10737418240)
    Alias       output_main
    Match       *
    Resource    gce_instance
    workers    10
```

[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [socket] could not validate socket status for #324 (don't worry)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [socket] could not validate socket status for #254 (don't worry)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [socket] could not validate socket status for #431 (don't worry)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [socket] could not validate socket status for #394 (don't worry)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [socket] could not validate socket status for #280 (don't worry)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)
[2022/03/04 17:47:05] [debug] [socket] could not validate socket status for #277 (don't worry)
[2022/03/04 17:47:05] [debug] [upstream] drop keepalive connection #-1 to logging.googleapis.com:443 (keepalive idle timeout)

**Comment by sophieyfang** [comment]
This is currently blocking https://github.com/GoogleCloudPlatform/ops-agent/pull/433

**Comment by wtchangdm** [comment]
We are facing this on 1.8.13 as well. Our environment is using simple `tail` input + `cloudwatch_logs` output.

Input bytes are less than 200KB/s and input records are ~200/s at most. Checked debug logs and haven't seen any warning messages yet.

I can attach configs and logs later today if needed.

**Comment by lecaros** [comment]
Hi @sophieyfang @wtchangdm  we've merged PR #5032 to fix this in the 1.8 branch.
Please let us know if you still face this issue after the patch.

**Comment by sophieyfang** [comment]
The com
